### PR TITLE
Condense parse_swift_recon

### PR DESCRIPTION
**Pull request by zreichert**

### CODE REVIEW

File: pytest_rpc/helpers.py [hunk 1]
```diff
@@ -501,24 +502,16 @@ def parse_swift_recon(recon_out):
     """
 
     lines = recon_out.splitlines()
+    delimiter_regex = re.compile('={20,}')
```
Comment (johnduarte) [inline pytest_rpc/helpers.py]: I would prefer to assume a full 80n character delimiter (79 `=` and a newline) to ensure that we are not caught by some other type of output.
Remote possibility:
```
Problem with host 3 <====================
```

**Review by johnduarte — CHANGES_REQUESTED** [review]

**Comment by johnduarte** [comment]
Otherwise, this is really awesome @zreichert . Thank you for cleaning this up for me.

File: pytest_rpc/helpers.py [hunk 2]
```diff
@@ -501,24 +502,16 @@ def parse_swift_recon(recon_out):
...
-            collection.append(data)
-            data = []
-            # Reset counter to use the delimeter to trigger data collection in
-            # the next pass.
-            i = i - 1
-        else:
-            data.append(lines[i])
-        i = i + 1
+
+    delimiter_positions = [ind for ind, x in enumerate(lines) if re.match(delimiter_regex, x)]
```
Comment (ryan-rs) [inline pytest_rpc/helpers.py:508]: I'm not sure how this even works. The `delimiter_regex` is a [RegexObject](https://docs.python.org/2/library/re.html#re.RegexObject), but [re.match](https://docs.python.org/2/library/re.html#re.match) is expecting a _regex pattern_ as a string.

Instead I suggest this list comprehension:

```python
delimiter_positions = [ind for ind, x in enumerate(lines) if delimiter_regex.match(x)]
```

File: pytest_rpc/helpers.py [hunk 3]
```diff
@@ -501,24 +502,16 @@ def parse_swift_recon(recon_out):
     """
 
     lines = recon_out.splitlines()
+    delimiter_regex = re.compile('^={79}')
```
Comment (ryan-rs) [inline pytest_rpc/helpers.py:505]: For safety use raw strings for regex patterns:

```python
delimiter_regex = re.compile(r'^={79}')
```

**Review by johnduarte — APPROVED** [review]

**Pull request merged**